tensorflow/python/ops/data_flow_ops.py

```diff
@@ -2183,12 +2190,26 @@ def __init__(self,
     self._name = name
 
   def get_yield_op(self):
-    """Add a node that yields a minibatch every time it is executed."""
-    return gen_data_flow_ops.record_input(
+    """Adds a node that yields a group of records every time it is executed.
+    If RecordInput `batches` parameter is not None, it yields a list of
+    record batches with the specified `batch_size`.
+    """
+    records = gen_data_flow_ops.record_input(
         file_pattern=self._file_pattern,
         file_buffer_size=self._buffer_size,
         file_parallelism=self._parallelism,
         file_shuffle_shift_ratio=self._shift_ratio,
         batch_size=self._batch_size,
         file_random_seed=self._seed,
         name=self._name)
+    if self._batches is None:
+      return records
+    else:
+      with ops.name_scope(self._name):
+        batch_list = [[] for i in six.moves.range(self._batches)]
+        records = array_ops.split(records, self._batch_size, 0)
+        records = [array_ops.reshape(record, []) for record in records]
+        for index, protobuf in zip(six.moves.range(len(records)), records):
+          batch_index = index % self._batches
+          batch_list[batch_index].append(protobuf)
```

Comment:
@ekelsen I just realized I forgot to include the equivalent of the lines that [parallel_stack batches](https://github.com/tensorflow/benchmarks/blob/b9221114d1261ec28f103455178abcfbeeb9b498/scripts/tf_cnn_benchmarks/preprocessing.py#L378) together, do you think that belongs here as well or should it stay in user code? 

Leaving things as they are may make sense in this case because this code contains full records they'll need to reorganize according to their features.

```
      for device_index in xrange(self.device_count):
        images[device_index] = tf.parallel_stack(images[device_index])
        label_index_batch[device_index] = tf.concat(labels[device_index], 0)
```